Upgrade React and Antd till the latest version

bsekachev opened the pull request:
<!-- Raise an issue to propose your change (https://github.com/opencv/cvat/issues).
It helps to avoid duplication of efforts from multiple independent contributors.
Discuss your ideas with maintainers to be sure that changes will be approved and merged.
Read the [Contribution guide](https://opencv.github.io/cvat/docs/contributing/). -->

<!-- Provide a general summary of your changes in the Title above -->

### Motivation and context

- [x] Annotation menu
- [x] OpenCV is not loaded automatically
- [x] Eslint
- [x] Stylelint
- [x] Tests


### How has this been tested?
<!-- Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc. -->

### Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply.
If an item isn't applicable for some reason, then ~~explicitly strikethrough~~ the whole
line. If you don't do that, GitHub will show incorrect progress for the pull request.
If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment <!-- see top comment in CHANGELOG.md -->
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

bsekachev commented:
/check

bsekachev commented:
/check

bsekachev commented:
/check

klakhov commented on cvat-ui/src/components/actions-menu/styles.scss:
```diff
@@ -1,22 +1,9 @@
-        width: 13em;
-    }
-}
+@import '../../base';
```
```suggestion
@import 'base';
```

klakhov commented on cvat-ui/src/components/actions-menu/styles.scss:
```diff
@@ -1,22 +1,9 @@
-}
+@import '../../base';
 
 .cvat-menu-dump-submenu-item,
```
I cant find usage of this classes in our code now

klakhov commented on cvat-ui/src/components/analytics-page/styles.scss:
```diff
@@ -129,7 +130,7 @@
     padding: 6px 8px;
     border: 1px solid lightgray;
     margin-left: 8px;
```
```suggestion
    margin-left: $grid-unit-size;
```

klakhov reviewed (commented):
Sidepanel became white:
![image](https://github.com/cvat-ai/cvat/assets/50956430/5a8b5b70-4087-4df2-8778-0170e7d0c56d)

klakhov reviewed (commented):
There is something wrong with initial position of opencv tool popup 
![opencvplacement](https://github.com/cvat-ai/cvat/assets/50956430/c455ccc6-e868-45b8-88bb-ce2663400ddd)

bsekachev commented:
@klakhov  Applied

klakhov reviewed (commented):
Exclamation mark and excessive left padding now appear on organization switch modal. Also I cant close it without selecting something new
![image](https://github.com/cvat-ai/cvat/assets/50956430/a00e724b-b774-4df3-a24e-77e788fa4af1)

Do we really want to use blue hover effect? I think it doesn't stand out enough on gray background. Maybe we can use scaling effect instead.
![image](https://github.com/cvat-ai/cvat/assets/50956430/289a8130-849c-4f94-aba4-4362238a5c7b)

Here I suppose using bold black font would be better instead of blue too. 
![image](https://github.com/cvat-ai/cvat/assets/50956430/9eb8f396-ec8c-4ba7-bf22-832fa634422a)

If we dont have an assignee, maybe we dont want to show the line with it at all? Or at least say something like not assigned so we dont have emty space.
I like the idea of showing stage+state in one status row like `annotation in progress` or `annotation rejected`, it looks cool. But it looks strange with `new`, maybe we should use an alias for that something like `annotation not started`. 
![image](https://github.com/cvat-ai/cvat/assets/50956430/7217ec7a-ce5a-4c1b-844e-8cb26e809518)

Small padding problem on models page(it seems the same problem is on the projects page):
![image](https://github.com/cvat-ai/cvat/assets/50956430/4a00c2fe-9712-45b6-a798-41917f94df10)

Major version update for package.json?

bsekachev commented:
> Exclamation mark and excessive left padding now appear on organization switch modal

I believe this is general antd change: 
![image](https://github.com/cvat-ai/cvat/assets/40690378/199e9334-5a81-4d21-bfa1-7af7077db916)

> Major version update for package.json?

I do not see reasons to update major version. Generally all the changes are cosmetic.

bsekachev commented:
>  Also I cant close it without selecting something new

Applied

> Do we really want to use blue hover effect?

Default antd button, I reversed it as it was with custom css.

> Here I suppose using bold black font would be better instead of blue too.

The same.

 > If we dont have an assignee, maybe we dont want to show the line with it at all? Or at least say something like not assigned so we dont have emty space.

The main idea is to keep height the same. Sliding bottom part is annoying a little bit. 

> I like the idea of showing stage+state in one status row like annotation in progress or annotation rejected, it looks cool. But it looks strange with new, maybe we should use an alias for that something like annotation not started.

The main idea behind it was to keep vertical space, using horizontal free space. I would avoid customizing logic in described way.

> Small padding problem on models page(it seems the same problem is on the projects page):

Fixed on projects/jobs/models. Also fixed top bars on different resolutions.

klakhov reviewed (approved):
Looks good!
